packages/react-dom/src/client/ReactDOM.js

```diff
@@ -736,7 +736,12 @@ const DOMRenderer = ReactFiberReconciler({
       oldText: string,
       newText: string,
     ): void {
-      textInstance.nodeValue = newText;
+      // IE9 will raise an exception if modifying a detached text node
+      // eslint-disable-next-line
+      // https://connect.microsoft.com/IE/feedbackdetail/view/944330/invalid-argument-error-when-changing-nodevalue-of-a-text-node-removed-by-setting-innerhtml-on-an-ancestor
+      if (textInstance.parentNode) {
+        textInstance.nodeValue = newText;
```

Comment:
The link is dead for me. Does accessing `parentNode` have an undue reflow effects? I don't have any reason to think it does but the DOM is weird :P

Reply:
> Microsoft Connect has been retired

_sigh_. I'll try to find it again...

Reply:
Does this actually fix the undo behavior, or just stop it from throwing? Since we're not setting `nodeValue` I'd expect the behavior to still be broken, no?

Reply:
It just stops it from throwing. I haven't been able to determine if react is incorrectly updating stale nodes and IE9 is just less lenient, or if this is specific to IE9.

I can dig in a bit more.

Reply:
I [noticed recently](https://github.com/facebook/react/issues/12420#issuecomment-375484576) that it's possible for Fibers to retain references to detached nodes. I wonder if this is potentially related? Does this issue occur in React 15.x?

Reply:
Dunno if it helps: https://github.com/facebook/react/issues/12420#issuecomment-377952155

Reply:
Back to the link issue. I'm having trouble digging up more information. For now, I've removed the link. I think the remaining comment is still sufficient.